src/main/java/com/iota/iri/LedgerValidator.java

```diff
@@ -220,6 +220,15 @@ private MilestoneViewModel buildSnapshot() throws Exception {
 
                     log.info(logMessage.toString());
                 }
+
+                // if we face a milestone that wasn't processed by updateSnapshot, correctly -> abort and let the
+                // "Solid Milestone Tracker" do it's magic
+                //
+                // NOTE: this can happen if a new subtangle becomes solid before a previous one while syncing
+                if(TransactionViewModel.fromHash(tangle, candidateMilestone.getHash()).snapshotIndex() != candidateMilestone.index()) {
```

Comment:
nitpicking:
I would say that the code will be clearer and easier to debug if it will be written in two lines:
`TransactionViewModel milestoneTx = TransactionViewModel.fromHash(tangle, candidateMilestone.getHash())`
`if (milestoneTx.snapshotIndex() != candidateMilestone.index())`